File: solutions/1332/main.py
class Solution:
    @staticmethod 
    def isPalindrome(s: str) -> bool: 
        i, j = 0, len(s)-1 
        while i < j: 
            if s[i] != s[j]: 
                return False 
            i += 1 
            j -= 1 
        return True  
    def removePalindromeSub(self, s: str) -> int:
        result = 0 
        while not self.isPalindrome(s) and len(s) > 0: 
            s = s.replace("a", "")
            result += 1
        result += 1
        return result

if __name__ == "__main__": 
    s = "baabb"
    sol = Solution()
    assert sol.removePalindromeSub(s) == 2

File: solutions/1332/main.cpp
/* ************************************************************************
> File Name:     main.cpp
> Author:        Sean
> Project:    LeetCode
> Created Time:  Sat Jan 22 09:01:30 2022
> Description:   
 ************************************************************************/
#include <string> 
#include <gtest/gtest.h> 

using namespace std; 

class Solution {
public:
    int removePalindromeSub(string s) {
        for (int i = 0; i < s.size(); ++i) {
            if (s[i] != s[s.size()-i-1]) return 2;
        }
        return 1;
    }
};

TEST(Test1332, SimpleTest) {
    string s = "baabb"; 
    Solution sol; 
    EXPECT_EQ(sol.removePalindromeSub(s), 2);
}

int main() {
    testing::InitGoogleTest();
    return RUN_ALL_TESTS();
}